### ISSUE TITLE
Build for "kccat6" broken

### ISSUE THREAD

**Issue by fwg-cag** [lineageos4microg/docker-lineage-cicd#94]
Installing the recent lineage-16.0-20200816-microG-kccat6.zip on an encrypted device leads to a bootloop: After entering the decryption code, the device just boots again.

Apparently, boot.img is somehow broken. Recovering an old boot.img to /boot on top after installing the image above results in a working device although on a previous kernel build.

Other Galaxy S5 builds for klte and klteduos are fine. 

Please fix this or advise if this is likely a problem with LineageOS itself?

**Comment by spolack** [comment]
If possible, could you please check if its similar behavior with a clean install? for now i'll withdraw the build.

**Comment by fwg-cag** [comment]
I can't do a full clean install as the device is a daily driver.

However, I
1. ran the update from the previous version and didn't get further than entering the decryption code before rebooting. Than, I 
2. adb sideloaded the zip file with cleaning dalvik & cache but ran into the same bootloop. 
Only an old boot.img from a backup got me further booting. However, the most recent boot.img applied to the previous system crashed as well after entering the decryption key...

BTW, thanks so much for reviving lineage4microg!!!

**Comment by spolack** [comment]
Ok what do you suggest in order to proceed? For now i'll blacklist that device unless someones willing to test a clean install.

**Comment by fwg-cag** [comment]
I propose to wait for the output of the next build cycle since there might also be something wrong with the original LOS source - I'll look out whether issues are reported there. 

As soon as the next lineage-16.0-202008*-microG-kccat6.zip is available, I will test it by updating the above device. If it fails, I may organise a second device to test a clean install...

So, remove it from the blacklist for this trial early next week?

**Comment by JuniorJPDJ** [comment]
@fwg-cag just do full twrp backup and try clean install

**Comment by fwg-cag** [comment]
I'll be happy to do that on the second test device with the next build, but won't play around with a daily driver ...

**Comment by spolack** [comment]
@fwg-cag  Could you please test with the new build?

**Comment by kurt-by** [comment]
There is no new build in https://download.lineage.microg.org/kccat6/
Maybe sumomogg @ xda-developers can test.

**Comment by motor-glider** [comment]
@spolack: Hello, is there a possibility to download the (probaby not functioning) version? I have a currently unused G901F I would like to install Lineage/MicroG on, and would be willing to help, if I can (some technical background is available: Shell, C, etc..., even Java, but I dislike it.). So it's in my very interest to make kccat6 work.

**Comment by fwg-cag** [comment]
I’ll do as soon as it is available at https://download.lineage.microg.org/kccat6/ …

**Comment by motor-glider** [comment]
System built fine, going to try it tonight.

**Comment by motor-glider** [comment]
I'm in a boot loop too.
I managed to once completely reinstall TWRP, then Lineage and Magisk and boot Lineage afterwards. System seemed to work, but after a shutdown/reboot cycle for inserting an SD card, the problem reoccured.
But: I don't treat that as a Lineage, but a TWRP problem. Most remarkable point is, that complete TWRP config is lost when first getting into the loop. Usually, I set timezone, language, timeout and disable f...ing vibration when TWRP is installed at first, all that is lost, when getting to the loop.
Could someone drop a note on where the meaning of different partitions in Samsung devices is documented, and how/where/when/why the bootloader may be overwritten?
Will retry with now inserted SD card.

**Comment by motor-glider** [comment]
There's some permission problem at the end of dmesg.log:
<5>[  127.199563] type=1400 audit(1598599364.068:227): avc:  denied  { write } for  pid=272 comm="recovery" name="0" dev="mmcblk0p27" ino=97730 scontext=u:r:recovery:s0 tcontext=u:object_r:media_rw_data_file:s0 tclass=dir permissive=1
<5>[  127.199645] type=1400 audit(1598599364.068:227): avc:  denied  { add_name } for  pid=272 comm="recovery" name="recovery.log" scontext=u:r:recovery:s0 tcontext=u:object_r:media_rw_data_file:s0 tclass=dir permissive=1
<5>[  127.199726] type=1400 audit(1598599364.068:227): avc:  denied  { create } for  pid=272 comm="recovery" name="recovery.log" scontext=u:r:recovery:s0 tcontext=u:object_r:media_rw_data_file:s0 tclass=file permissive=1
<5>[  127.202993] type=1400 audit(1598599364.068:227): avc:  denied  { write } for  pid=272 comm="recovery" path="/data/media/0/recovery.log" dev="mmcblk0p27" ino=97756 scontext=u:r:recovery:s0 tcontext=u:object_r:media_rw_data_file:s0 tclass=file permissive=1
<5>[  127.203097] type=1300 audit(1598599364.068:227): arch=40000028 syscall=322 per=800008 success=yes exit=22 a0=ffffff9c a1=a7b79fc0 a2=20241 a3=1b6 items=2 ppid=1 ppcomm=init pid=272 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 ses=4294967295 tty=(none) comm="recovery" exe="/sbin/recovery" subj=u:r:recovery:s0 key=(null)
<5>[  127.203206] type=1307 audit(1598599364.068:227):  cwd="/"
<5>[  127.203242] type=1302 audit(1598599364.068:227): item=0 name="/data/media/0/" inode=97730 dev=b3:1b mode=040770 ouid=1023 ogid=1023 rdev=00:00 obj=u:object_r:media_rw_data_file:s0 nametype=PARENT
<5>[  127.203313] type=1302 audit(1598599364.068:227): item=1 name="/data/media/0/recovery.log" inode=97756 dev=b3:1b mode=0100666 ouid=0 ogid=0 rdev=00:00 obj=u:object_r:media_rw_data_file:s0 nametype=CREATE
<5>[  127.203383] type=1327 audit(1598599364.068:227): proctitle="/sbin/recovery"
<5>[  127.204254] type=1400 audit(1598599364.076:228): avc:  denied  { setattr } for  pid=272 comm="recovery" name="recovery.log" dev="mmcblk0p27" ino=97756 scontext=u:r:recovery:s0 tcontext=u:object_r:media_rw_data_file:s0 tclass=file permissive=1

**Comment by motor-glider** [comment]
And... there's a kernel panic in the logs. So most probably some kernel setting is wrong. Details can be provided if someone is interested.

**Comment by kurt-by** [comment]
There is a similar Problem with official lineageOS and a don't see an issue for that in https://gitlab.com/search?group_id=3936439&project_id=&repository_ref=&scope=issues&search=kccat6&snippets=false

Maybe you should file a bug report @motor-glider

**Comment by motor-glider** [comment]
Indeed - just tried it here... and I should file that. Will check when it was introduced if I can get old enough zip files.

**Comment by motor-glider** [comment]
https://gitlab.com/LineageOS/issues/android/-/issues/2432 - hope someone will at least have a look on that.

**Comment by motor-glider** [comment]
Just managed to get out of the bootloop.

After inserting a really slow SD card and reflashing stock Lineage, testing, doing backup, and finally flashing Lineage for MicroG, I got into the reboot loop. Reboot repeatedly appeared just after the system tried to mount SD card (unencrypted), and I was able to recover with the boot image from the above mentioned Stock Lineage backup.

**Comment by motor-glider** [comment]
comparing both boot.img from Stock Lineage and Lineage for MicroG, the following looks interesting:
* ramdisks seem to differ only in build time,
* kernels differ,
* extracted kernel configs look the same.
So IMHO there could be a problem with the toolset in the docker image.

**Comment by spolack** [comment]
I'll try later this week or next week to rework the build process and align it to official instructions. This will hopefully fix that issue.

**Comment by fwg-cag** [comment]
@spolack: Any chance for a new trial anytime soon?

**Comment by motor-glider** [comment]
Just going to do, thanks for the reminder.

**Comment by cyBea** [comment]
I also have this sort of boot loop using lineage-16.0-20200816-microG-kccat6.zip. I "fixed" it by removing sd card which is formatted as "internal". But like this I have VERY reduced functionality.
Can I help providing logs?

**Comment by fwg-cag** [comment]
@spolack: Thanks for the new 17.1 builds - now I'll be curious about the 16.0 builds once triggered by you ;-)

**Comment by motor-glider** [comment]
Sorry for the late response - but still in boot loop.

**Comment by kurt-by** [comment]
@motor-glider are you talking about today's build https://download.lineage.microg.org/kccat6/lineage-16.0-20200928-microG-kccat6.zip ?

**Comment by motor-glider** [comment]
No, forget about that, did my own build with fresh sync from all git repos.

**Comment by fwg-cag** [comment]
@motor-glider: I tried the today's build anyway https://download.lineage.microg.org/kccat6/lineage-16.0-20200928-microG-kccat6.zip and found out:
- OTA updating an encyrpted device results in a bootloop,
- Clean install works until encryption is enabled - next boot results in a bootloop
*The problem here appears to be a broken boot.img which is unable to cope with device encryption.* As soon as I replace the boot.img, e.g. with an old one or even with the most recent original lineage-16.0-20200924-nightly-kccat6-signed.zip, all is fine... As lineage itself is fine, I assume you indeed need a fresh sync from the boot.img relevant repos...
I look forward to learn about your own build ...

**Comment by motor-glider** [comment]
@kurt-by and @fwg-cag: Just tried the build from the same URL. Following observations, maybe something's wrong with my device only:
* booting without any SD card works as expected.
* booting with any formatted SD card leads to bootloop (tried 2 different ones).
* booting with SD card with emptied out partition table (GPT) works, but
* formatting SD card reboots early (20 %) and goes into bootloop.
So IMHO something is wrong with the card's SD code. Will extract the kernel and config from there, hope I'll find something... but not before tomorrow.

**Comment by motor-glider** [comment]
One more experience:
* if I try to format the card as Mobile Memory (didn't try with english language, in german: Mobiler Speicher), there's an immediate return with a message that SD is ready to use, but that's simply not true. Neither any subdevice (partition od /dev/block/mmcblk1) comes up, nor any change happens in displayed memory size, while the status message "SD card from SanDisk is not supported" is kept. Formatting as device memory leads to the already mentioned boot loop.

Will redo al sync tomorrow, have it scripted a bit... :)

**Comment by fwg-cag** [comment]
Addendum: The working boot.img from the most recent original lineage-16.0-20200924-nightly-kccat6-signed.zip shows version 3.10.108 while here 3.4.113 appears to prevail. Is this another hint for some wrangled repo?

**Comment by bege10** [comment]
Yesterday I installed the new update from 20201020 on a not encrypted phone and by now it works. I only had to reinstall one app that did not start any more.

**Comment by bege10** [comment]
LineageOS for kccat6 has been updated at November 5. The LineageOS for microG update is missing.

**Comment by kurt-by** [comment]
November 12 has just been built.

**Comment by spolack** [comment]
Is the build still broken?

**Comment by motor-glider** [comment]
For me, it is. Reboot occurs when trying to mount encrypted SD card. This happens both with downloaded image as well as with own builds. Still trying to find out more from time to time, but it's not easy to get logs from that early stage of system startup.

**Comment by bege10** [comment]
The December build works for me, but **no** encrypted data.
The January build corrupts root with addon-su here.

I now little about system issues. Regarding boot logs, do you know this information? It talks about boot log via ADB or script (script added manually or via app). I used the linked TeMeFI app to get a boot log for a Magisk issue.
https://www.didgeridoohan.com/magisk/MagiskHelp
https://forum.xda-developers.com/android/apps-games/app-temefi-comprehensive-documentation-t3806260

**Comment by spolack** [comment]
> For me, it is. Reboot occurs when trying to mount encrypted SD card. This happens both with downloaded image as well as with own builds. Still trying to find out more from time to time, but it's not easy to get logs from that early stage of system startup.

Is the behavior different from official lineageos?

**Comment by motor-glider** [comment]
@bege10: Thanks for the links, will check them. Tried to hook up some own code, but in that early stage of boot, it's not quite easy (and I'm only familiar a bit with regular linux, but not really with Android).
@spolack: Just tried the december 31 lineageos build, it seems to work fine. Is there any good documentation or description on what MicroG is and how it works (or could you or someone else explain it to me, maybe per mail, as that would overstress that forum)?

**Comment by fwg-cag** [comment]
@spolack:

I have tried the lineage-16.0-20210112-microG-kccat6.zip build on 2 encrypted phones without SD card. After the update the phones boot to the point at which the PIN to release encryption has to be entered, but after entering the PIN the phones reboot again.

> Is the behavior different from official lineageos?
Yes it is. This behaviour doesn't appear with the official lineageos builds.

Moreover, when I overwrite boot.img after updating to lineage-16.0-20210112-microG-kccat6.zip with the one extracted from the recently original lineageos build, the phones work smoothly on microg-lineage. As I described above, in lineage-16.0-20210112-microG-kccat6.zip (only) the boot.img appears to be broken.
